### ISSUE TITLE
CredScan as a pre-commit git hook

### ISSUE THREAD

**Issue by nikithauc** [microsoftgraph/msgraph-sdk-javascript#731]

**Comment by MIchaelMainer** [comment]
PR is too late, secrets are already pushed to public, threat actors may have already consumed them. This is a dev setup thing. Now to think of it, not really a work item JS repo.

**Comment by zengin** [comment]
@MIchaelMainer we are adding it as a gate too in Raptor in case the secrets slip through from the dev setup: https://github.com/microsoftgraph/msgraph-sdk-raptor/pull/955.

**Comment by MIchaelMainer** [comment]
Yes, as a gate, that's good. I still want everyone running CredScan locally. ☺️

**Comment by nikithauc** [comment]
pre-commit hook + gate then?

**Comment by zengin** [comment]
I'd say yes. One is to prevent it from happening, the other is to catch if it happens. I think we have a way to catch it after the check-in by default security scanning tools, but the earlier we catch, the better.

**Comment by zengin** [comment]
That said, as @MIchaelMainer said, "pre-commit hook" is not really a work item specific to the repo.

**Comment by MIchaelMainer** [comment]
@zengin Yes, we should also have this as PR gate.
